app/controllers/status_controller.rb

```diff
@@ -1,8 +1,26 @@
 # frozen_string_literal: true
 
+require 'date'
+
 class StatusController < ApplicationController
+  include TokenUser
+
   def index
     render 'index', layout: false
   end
 
+  # /whoami returns basic information about the request client and access token
+  # that Conjur receives.
+  #
+  # This is useful for troubleshooting authentication with access tokens and
+  # configuring proxies or load balancers.
+  def whoami
+    render json: { 
+      client_ip: request.ip,
+      user_agent: request.user_agent,
+      account: token_user.account,
+      username: token_user.login,
+      token_issued_at: Time.at(token_user.token.claims["iat"])
```

Comment:
Now authentication is required for this endpoint and there is no way for tempering and generating a JWT that contains illegal formating of `iat` that might cause `Time.at` to raise exception right?

Reply:
That's right. The validation itself uses `Time.at`. If that raises an exception, that validation is going to fail, never reaching our use of the claim.

See: https://github.com/cyberark/slosilo/blob/c2bb674c75c1a6ff2af66dfb5da68cb433dcc5f4/lib/slosilo/key.rb#L148